core/src/test/java/io/parsingdata/metal/SubStructTest.java

```diff
@@ -133,15 +133,15 @@ public void linkedListWithCycleToLowerOffset() throws IOException {
     }
 
     private void checkBranch(final ParseGraph graph, final int graphOffset, final int nextOffset) {
-        checkValue(graph.head, 1, graphOffset + 2); // footer
+        checkValue(graph.head.asGraph().head, 1, graphOffset + 2); // footer
         checkValue(graph.tail.tail.head, nextOffset, graphOffset + 1); // next
-        checkValue(graph.tail.tail.tail.head, 0, graphOffset); // header
+        checkValue(graph.tail.tail.tail.head.asGraph().head, 0, graphOffset); // header
```

Comment:
These `graph.tail.tail.tail.head` tests are very hard to review :stuck_out_tongue_closed_eyes: